nixos/modules/services/databases/postgresql.nix

```
@@ -532,6 +596,13 @@ in
             done
 
             if test -e "${cfg.dataDir}/.first_startup"; then
+              if test -e "${cfg.dataDir}/.post_upgrade"; then
+                ${optionalString cfg.analyzeAfterUpgrade ''
+                  vacuumdb --port=${toString cfg.port} --all --analyze-in-stages
+                ''}
+                rm "${cfg.dataDir}/.post_upgrade"
```

```suggestion
                rm -f "${cfg.dataDir}/.post_upgrade"
```
don't fail if that somehow vanished